tests/testthat/test-util.R

```diff
@@ -4,15 +4,29 @@ test_that("select_assay selects active assay", {
   rna3 <- create_count_mat(5, 5)
 
   obj <- Seurat::CreateSeuratObject(rna1, assay="rna1")
-  obj[["rna2_"]] = Seurat::CreateAssayObject(rna2, key="rna2_")
-  obj[["rna3_"]] = Seurat::CreateAssayObject(rna3, key="rna3_")
+  obj[["rna2"]] = Seurat::CreateAssayObject(rna2, key="rna2_")
+  obj[["rna3"]] = Seurat::CreateAssayObject(rna3, key="rna3_")
 
   expect_equal(Seurat::DefaultAssay(object = obj), "rna1")
-  Seurat::DefaultAssay(object = obj) <- "rna2_"
-  expect_equal(Seurat::DefaultAssay(object = obj), "rna2_")
+  Seurat::DefaultAssay(object = obj) <- "rna2"
+  expect_equal(Seurat::DefaultAssay(object = obj), "rna2")
 
   assay <- select_assay(obj)
-  expect_equal(names(assay), "rna2_")
+  expect_equal(names(assay), "rna2")
+})
+
+test_that("counts_matrix_from_assay works on different assay version", {
+  rna1 <- create_count_mat(5, 5)
+  rna2 <- create_count_mat(5, 5)
+  rna3 <- create_count_mat(5, 5)
+
+  obj <- Seurat::CreateSeuratObject(rna1, assay="rna1")
+  obj[["rna2"]] = Seurat::CreateAssayObject(rna2, key="rna2_")
+  obj[["rna3"]] = SeuratObject::CreateAssay5Object(rna3, key="rna3_")
```

Comment:
```suggestion
  rna <- create_count_mat(5, 5)

  obj <- Seurat::CreateSeuratObject(rna, assay="rna1")
  obj[["rna2"]] = Seurat::CreateAssayObject(rna, key="rna2_")
  obj[["rna3"]] = SeuratObject::CreateAssay5Object(rna, key="rna3_")
```

R will pass function arguments by value rather than reference so you shouldn't need to construct multiple instances yourself

Reply:
I actually want to create three different matrices because I want their values to be different for the tests.  Turns out R is copy by value, but only if the value is modified.